cpp/src/arrow/compute/kernels/scalar_set_lookup.cc

```diff
@@ -54,8 +61,8 @@ struct SetLookupState : public KernelState {
     } else {
       return Status::Invalid("value_set should be an array or chunked array");
     }
-    if (!options.skip_nulls && lookup_table.GetNull() >= 0) {
-      null_index = memo_index_to_value_index[lookup_table.GetNull()];
+    if (!options.skip_nulls && lookup_table.value().GetNull() >= 0) {
+      null_index = memo_index_to_value_index[lookup_table.value().GetNull()];
```

Comment:
Can probably use the dereference operator?
```suggestion
    if (!options.skip_nulls && lookup_table->GetNull() >= 0) {
      null_index = memo_index_to_value_index[lookup_table->GetNull()];
```